src/commands/convertConnections/ConvertSettingsStep.ts

```diff
@@ -0,0 +1,236 @@
+                                label: localize('addedSetting', `Add Local setting "${connection.name}"`),
+                                iconPath: activitySuccessIcon
+                            })
+                        );
+                    }
+                    await AzExtFsExtra.writeJSON(nonNullProp(context, 'localSettingsPath'), localSettings);
+                    await ext.rgApi.workspaceResourceTree.refresh(context);
+                }
+            } else {
+                if (context.functionapp) {
+                    const client = await context.functionapp?.site.createClient(context);
+                    const remoteSettings = await client.listApplicationSettings();
+                    for (const connection of context.convertedConnections) {
+                        if (remoteSettings.properties) {
+                            remoteSettings.properties[connection.name] = connection.value;
+                        } else {
+                            await client.updateApplicationSettings({ properties: { [connection.name]: connection.value } });
+                        }
+                        context.activityChildren?.push(
+                            new GenericTreeItem(undefined, {
+                                contextValue: createUniversallyUniqueContextValue(['useExistingResourceGroupInfoItem', activitySuccessContext]),
+                                label: localize('addedSetting', `Add app setting "${connection.name}" `),
+                                iconPath: activitySuccessIcon
+                            })
+                        );
+                    }
+                    //may need to add settings to include the clientId and configuration properties
+                    await client.updateApplicationSettings({ properties: remoteSettings.properties });
+                    await ext.rgApi.tree.refresh(context);
+                }
+            }
+        }
+    }
+
+    public shouldExecute(context: IConvertConnectionsContext): boolean {
+        return !!context.connections;
+    }
+}
+
+async function getScopeHelper(context: IConvertConnectionsContext, accountName: string, filter: string): Promise<string> {
+    const client = await createResourceClient(context);
+    const resources = await uiUtils.listAllIterator(client.resources.list({ filter }));
+
+    for await (const resource of resources) {
+        if (resource.name === accountName) {
+            return nonNullProp(resource, 'id');
+        }
+    }
+
+    throw new Error(localize('noResourceFound', `No resource found with name "${accountName}" in subscription "${context.subscriptionDisplayName}"`));
+}
+
+function getClientIdAndCredentialProperties(context: IConvertConnectionsContext, connectionName: string): Connection[] {
+    const clientIdAndConfigurationProperties: Connection[] = [];
+    if (!context.local) {
```

Comment:
This is something that could go into a `RemoteSettingsConvertStep` if you factor it out.